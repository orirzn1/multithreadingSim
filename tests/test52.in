L11
S8
O12
N9

T0
I@0x00000000
ADDI $4, $1, 0x1
LOAD $1, $4, 0x0
SUB $3, $5, $0
ADDI $6, $4, 0x4
ADD $0, $1, $2
HALT $0

T1
I@0x00000000
ADDI $6, $7, 0x4
LOAD $4, $1, 0x0
SUB $1, $3, $0
SUB $4, $3, $3
ADD $2, $4, $3
ADDI $0, $3, 0x4
ADDI $1, $7, 0x3
ADD $2, $2, $2
HALT $0

T2
I@0x00000000
STORE $1, $6, 0x2
ADDI $3, $3, 0x4
STORE $5, $3, 0x0
ADD $3, $2, $5
STORE $0, $2, $3
ADD $2, $4, $6
SUBI $6, $3, 0x1
STORE $1, $0, $1
LOAD $7, $7, $7
ADD $6, $3, $1
HALT $0

T3
I@0x00000000
ADD $1, $4, $7
SUB $1, $2, $3
LOAD $7, $0, 0x4
SUB $5, $6, $1
SUB $3, $4, $0
STORE $7, $6, 0x0
STORE $1, $6, 0x2
ADDI $7, $3, 0x3
HALT $0

T4
I@0x00000000
SUBI $3, $4, 0x2
SUB $2, $6, $6
STORE $3, $1, 0x0
ADDI $2, $7, 0x2
SUB $1, $4, $1
HALT $0

T5
I@0x00000000
SUBI $4, $0, 0x3
SUBI $7, $7, 0x2
ADDI $1, $2, 0x2
SUBI $6, $1, 0x1
HALT $0

T6
I@0x00000000
LOAD $7, $2, $5
SUB $1, $3, $1
STORE $1, $6, 0x0
SUB $2, $0, $7
HALT $0

T7
I@0x00000000
SUB $2, $7, $7
LOAD $5, $6, $4
SUBI $4, $3, 0x2
SUBI $3, $7, 0x2
SUBI $3, $6, 0x1
ADD $2, $4, $4
SUB $6, $3, $4
ADD $0, $0, $7
HALT $0

T8
I@0x00000000
LOAD $4, $6, 0x4
LOAD $1, $4, 0x3
STORE $6, $2, 0x1
SUBI $0, $0, 0x2
STORE $2, $5, $3
HALT $0

D@0x0
0xA
0x2
0x5
0x7
0x6
0x4
0x6
0x9
0x9
