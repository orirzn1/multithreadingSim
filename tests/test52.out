
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0xA	R1 = 0xA	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 1:
	R0 = 0x4	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x1	R7 = 0xFFFFFFFE
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFE	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 5.727273

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 1:
	R0 = 0x4	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x1	R7 = 0xFFFFFFFE
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFE	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFE	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.212121

